lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -877,4 +878,35 @@ def (c, d) = ['fourth']
 return [a, b, c, d].join(' ')
 ''') == "first second fourth null"
     }
-}
+
+    @Issue("JENKINS-47363")
+    @Test
+    void excessiveListElements() {
+        def l = 0..250
+        def s = l.join(",\n")
+        try {
+            assert evalCPS("""
+def b = [${s}]
```

Comment:
I would recommend first asserting that you _can_ successfully create a list of 250 elements (or even of every size up to and including 250); and only then verifying that an attempt to make one of 251 elements fails gracefully.